python/cudf/cudf/core/column/column.py

```diff
@@ -321,22 +323,22 @@ def _fill(
         if end <= begin or begin >= self.size:
             return self if inplace else self.copy()
 
-        fill_scalar = as_device_scalar(fill_value, self.dtype)
+        slr = cudf.Scalar(fill_value, dtype=self.dtype)
```

Comment:
These changes are unrelated to the rest of the PR right?

Reply:
`__setitem__` uses `fill` when value is a scalar. I think the changes here are introduced to handle some corner cases of null/NA related issues. But looking at the factory function I'm not quite sure why they are needed. I'm rerunning the tests to make sure and I'll add a note here to clarify.

Reply:
See Brandon's comment: https://github.com/rapidsai/cudf/pull/10202#discussion_r806930768